### PR TITLE
verify and fix info schema tables - 2

### PR DESCRIPTION
**Pull request by jennifersp**
- Complete tables:
`CHARACTER_SETS`
`COLLATIONS` 
`ENGINES` - only the InnoDB engine is supported
`SCHEMATA`
`COLLATION_CHARACTER_SET_APPLICABILITY`
- Some info missing:
`COLUMN_STATISTICS` -  `histogram` json column only has `buckets` key values.
`PROCESSLIST` -  `status` is not defined completely; `"Query" command` is supported only.

### CODE REVIEW

File: sql/information_schema/information_schema.go [hunk 1]
```diff
@@ -887,6 +892,10 @@ func columnStatisticsRowIter(ctx *Context, c Catalog) (RowIter, error) {
 			}
 
 			for _, col := range t.Schema() {
+				privSetCol := privSetTbl.Column(col.Name)
+				if privSetCount == 0 && privSetDb.Count() == 0 && privSetTbl.Count() == 0 && privSetCol.Count() == 0 {
+					continue
+				}
```
Comment (fulghum) [inline sql/information_schema/information_schema.go:898]: Would be nice to have a simple `UserPrivilegeTest` to assert that when there are no privileges, a user can't see info they shouldn't see in `information_schema.column_statistics`.

**Review by fulghum — APPROVED** [review]
Looks good! 🚢 

Probably smart to have @Hydrocharged take a quick peek at the `collations.go` updates, too, just in case he sees something problematic that we missed.

File: sql/collations.go [hunk 2]
```diff
@@ -152,8 +152,8 @@ const (
 	Collation_ucs2_lithuanian_ci          CollationID = 140
 	Collation_ucs2_persian_ci             CollationID = 144
 	Collation_ucs2_polish_ci              CollationID = 133
-	Collation_ucs2_roman_ci               CollationID = 143
 	Collation_ucs2_romanian_ci            CollationID = 131
+	Collation_ucs2_roman_ci               CollationID = 143
```
Comment (Hydrocharged) [inline sql/collations.go:156]: I see that we added new collations, but why were the pre-existing collations moved around?

File: sql/mysql_db/privileged_database_provider.go [hunk 3]
```diff
@@ -157,15 +157,15 @@ func (pdb PrivilegedDatabase) GetTableNames(ctx *sql.Context) ([]string, error)
 		privSet := pdb.grantTables.UserActivePrivilegeSet(ctx)
 		dbSet := privSet.Database(pdb.db.Name())
 		// If there are no usable privileges for this database then no table is accessible.
-		if privSet.Count() == 0 && !dbSet.HasPrivileges() {
+		privSetCount := privSet.Count()
```
Comment (Hydrocharged) [inline sql/mysql_db/privileged_database_provider.go:160]: (Personal Preference) We can probably revert this since we're not reusing the count

**Review by Hydrocharged — APPROVED** [review]

File: sql/mysql_db/privileged_database_provider.go (hunk 3 above)
Comment (jennifersp) [reply sql/mysql_db/privileged_database_provider.go:160]: We're reusing the count on line 173

File: sql/collations.go (hunk 2 above)
Comment (jennifersp) [reply sql/collations.go:156]: Ah forgot to put back the moved around ones when I was matching it with MySQL result to track each of them.

File: sql/mysql_db/privileged_database_provider.go (hunk 3 above)
Comment (Hydrocharged) [reply sql/mysql_db/privileged_database_provider.go:160]: I see, I missed the deletion on line 168 XD my bad

**Pull request merged**